src/main/java/com/teragrep/pth_06/planner/walker/conditions/IndexCondition.java

```diff
@@ -0,0 +1,93 @@
+ * along with this program.  If not, see <https://github.com/teragrep/teragrep/blob/main/LICENSE>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+
+package com.teragrep.pth_06.planner.walker.conditions;
+
+import com.teragrep.pth_06.planner.StreamDBClient;
+import org.jooq.Condition;
+import org.w3c.dom.Element;
+
+import static com.teragrep.pth_06.jooq.generated.streamdb.Streamdb.STREAMDB;
+
+public final class IndexCondition implements QueryCondition {
+    private final boolean streamQuery;
+    private final Element element;
+
+    public IndexCondition(Element element, boolean streamQuery) {
+        this.streamQuery = streamQuery;
+        this.element = element;
+    }
+
+    public Condition condition() {
+        final String value = element.getAttribute("value");
+        final String operation = element.getAttribute("operation");
+        Condition condition;
+        if (streamQuery) {
+            condition =
+                    STREAMDB.STREAM.DIRECTORY.like(
+                            value.replace('*', '%')
+                    );
+        } else {
+            condition =
+                    StreamDBClient.GetArchivedObjectsFilterTable.directory.like(
+                            value.replace('*', '%').toLowerCase()
+                    );
+        }
+        if (operation.equalsIgnoreCase("NOT_EQUALS")) {
+            condition = condition.not();
+        }
+        return condition;
+    }
+
+    @Override
+    public boolean equals(final Object object) {
+        if (this == object) return true;
+        if (object == null) return false;
+        if (object.getClass() != this.getClass()) return false;
+        final IndexCondition cast = (IndexCondition) object;
+        return this.condition().toString().equals(cast.condition().toString());
```

Comment:
compare uses .toString() however toString is not implemented, it must use the object properties that make it equal to another i.e. in this case element and config

Reply:
this still contains toString